test/unit/javascript/fetchButtonDirectiveTests.js

```diff
@@ -96,6 +96,19 @@ describe('fetchButton', function() {
     it('should have the correct scope when successful with showSummaryStats', function() {
         _prepareScope(1, [1,2], true, {foo: {concepts: ['concept'], valid: true}});
         _clickButton('resolve()', 'resolve({result: {allSamples: 1337, subsets: null}})');
+        expect(element.find('span').text()).toBe('Task complete! Go to the "Run Analysis" tab to continue.');
+        expect(element.isolateScope().running).toBe(false);
+        expect(element.isolateScope().loaded).toBe(true);
+        expect(element.isolateScope().allSamples).toEqual(1337);
+    });
+
+    it('should show `Task complete! Go to the "Preprocess" or "Run Analysis" tab to continue.`', function() {
+        // NOTE overriding parts of `beforeEach` here
+        var html = '<fetch-button concept-map="conceptMap" allowed-cohorts="allowedCohorts" has-preprocess-tab="true"></fetch-button>';
```

Comment:
1. Der einzige Grund, warum es im rootScope gesetzt werden muss, ist dass diese beiden variablen nicht optional sind. D.h. wenn du sie beim Erstellen des elements nicht gesetzt hast, kracht es. Da hasPreprocessTab optional ist, kannst du den $rootScope ignorieren und musst auch den HTML snippet nicht veraendern. Hab eben am Telefon nicht daran gedacht.

2. Du erweiterst die Parameter um hasPreprocessTab und checkst ob es definiert ist, ansonsten setzt du es auf false. Was du in 3. geschrieben hast geht in die richtige Richtung, aber die Condition macht mir Gaensehaut. Als Daumenregel: Benutze kein != oder == in JS. Gib mal null == undefined in die Console ein. 
Mach lieber typeof hasPreprocessTab === 'undefined'